pythonbits/scene.py

```diff
@@ -1,81 +1,383 @@
-import progressbar
 from base64 import b64decode
-from zlib import crc32
+from bs4 import BeautifulSoup
+import re
+import guessit
+import functools
+import pprint
 
 from .logging import log
 
+
 srrdb = b64decode('aHR0cHM6Ly9zcnJkYi5jb20v').decode('utf8')
 
 
-def check_scene_rename(fname, release):
-    release_url = srrdb + "release/details/{}".format(release)
-    r = requests.get(release_url)
-
-    if fname not in r.text:
-        log.warning('Possibly renamed scene file!\n'
-                    '\tFilename {}\n\tnot found at {}',
-                    fname, release_url)
-
-
-def crc(path):
-    log.debug('Calculating CRC32 value')
-    checksum = 0
-    fsize = os.path.getsize(path)
-    i = 0
-    chunk_size = 4 * 2**20
-    with open(path, 'rb') as f:
-        with progressbar.DataTransferBar(max_value=fsize,
-                                         max_error=False) as bar:
-            while True:
-                i += 1
-                data = f.read(chunk_size)
-                if not data:
-                    return checksum & 0xFFFFFFFF
-
-                bar.update(i*chunk_size)
-                checksum = crc32(data, checksum)
-
-
-def is_scene_crc(path):
-    checksum = crc(path)
-    log.debug('CRC32 {:08X}', checksum)
-    r = requests.get(srrdb + 'api/search/archive-crc:%08X' % checksum)
-    r.raise_for_status()
-
-    scene = int(r.json()['resultsCount']) != 0
-    if int(r.json()['resultsCount']) > 1:
-        log.warning('More than one srrDB result for CRC32 query')
-    log.info('Scene checkbox set to {} '
-             'due to CRC query result'.format(scene))
-
-    if scene:
-        release = r.json()['results'][0]['release']
-        fname = os.path.basename(path)
-        check_scene_rename(fname, release)
-
-    return scene
-
-
-def query_scene_fname(path):
-    if os.path.isfile(path):
-        query = os.path.splitext(os.path.basename(path))[0]
-    elif os.path.isdir(path):
-        query = os.path.basename(path)
-    elif not os.path.exists(path):
-        raise FileNotFoundError('File or directory not found: %s' % (path,))
+class SceneError(Exception):
+    pass
+
+
+class APIError(Exception):
+    pass
+
+
+# FIXME: "hd1080-wtl.mkv" (from Walk.the.Line.Extended.Cut.2005.1080p.BluRay.x264-HD1080)
+#        should report an error, but it's not found on any scene indexers.
+
+
+def check_integrity(path, on_error=None):
+    """
+    Check if `path` (file or directory) was modified
+
+    Files that are not part of a scene release are ignored.
+
+    Return True/False if this release was (not) modified or None if not sure.
+    """
+    errors_found = False
+    something_found = False
+    if _isdir(path):
+        dirname = os.path.basename(path)
+    else:
+        dirname = ''
+    log.debug('Base path for {}: {}', path, dirname)
+
+    guess = _guessit(path)
+    if 'release_group' not in guess:
+        log.debug('No release group found - cannot find scene release: {}', guess)
+        return None
```

Comment:
So now the scene check completely fails when you have a renamed file with the group removed. This is a regression. It's also meant to help people discover scene files that they in fact have but that may have been removed due to some renaming script.

Reply:
What do you suggest? You can't really search for a release without the group and always asking the user is annoying.

To be honest, I've come to the conclusion that the scene rule is annoying. I've seen so many existing renamed and/or modified scene releases. I'm not sure if I really care about this pull request anymore. Nobody else seems to care about this tag.